src/lib.rs

```diff
@@ -170,8 +291,12 @@ impl StringSlice for str {
 
 #[cfg(test)]
 mod tests {
+    #[cfg(not(feature = "std"))]
     use core::ops::Bound;
 
+    #[cfg(feature = "std")]
+    use std::ops::Bound;
```

Comment:
```suggestion
    use core::ops::Bound;
```

Same as before, we should be able to use `core::*`